src/routes/auth/guards/auth.guard.ts

```diff
@@ -6,12 +6,22 @@ import {
 } from '@nestjs/common';
 import { IAuthRepository } from '@/domain/auth/auth.repository.interface';
 import { AuthController } from '@/routes/auth/auth.controller';
+import { AuthPayload } from '@/domain/auth/entities/auth-payload.entity';
+import { Request } from 'express';
+
+declare module 'express' {
+  // Inject AuthPayload to express.Request
+  interface Request {
+    auth?: AuthPayload | undefined;
+  }
+}
```

Comment:
What is the advantage of doing it this way? Wouldn't it be simpler to access it via `request['auth']`?

Reply:
This adds the property to the global types so it can be accessed in other files.

Reply:
But wouldn't that be done via the decorator?

Reply:
Would you prefer the type declaration be moved to the decorator? It is not possible to access the property unless it exists on the type:

> Element implicitly has an 'any' type because expression of type '"accessToken"' can't be used to index type 'Request<ParamsDictionary, any, any, ParsedQs, Record<string, any>>'.
  Property 'accessToken' does not exist on type 'Request<ParamsDictionary, any, any, ParsedQs, Record<string, any>>'.ts(7053)

![image](https://github.com/safe-global/safe-client-gateway/assets/20442784/0b675417-6c2d-4358-92d9-5fb29e92e3e4)

![image](https://github.com/safe-global/safe-client-gateway/assets/20442784/da4a1813-0a8c-48ce-bed4-6d14c7c49e43)

Reply:
I thought you could access the property without redeclaring the module. In that case I think it should be ok 👍